engine/src/main/java/org/terasology/config/flexible/SettingImpl.java

```diff
@@ -101,13 +104,13 @@ public T getValue() {
     }
 
     public boolean setValue(T newValue) {
-        if (!validator.validate(newValue))
+        if (!validator.validate(newValue)) {
+            LOGGER.warn("The passed value {} is invalid.", newValue);
```

Comment:
@emanuele3d but what if someone doesn't read the method doc before implementing it in their implementation of `SettingValueValidator` and does not log a warning? Instead, we can add another method to `SettingValueValidator` called `validateWithWarning` (or something better) which first returns whether the value is valid while logging an error if it is invalid, like follows:
```java
public boolean validateWithWarning(T value) {
    boolean isValid = this.validate(value);
    if (!isValid) {
        logger.warn("[Why the value is invalid]");
    }
    return isValid;
}
```